Enforce minimum interval between add-on repository updates

yol opened the pull request:
## Description
Repository update can fail in a way that would lead to an immediate reschedule of the update. A common reason would be a failure to update the `nextcheck` column of the `repo` table. This leads to mirror and repository servers getting constantly hammered.

Since there is no reason to have a smaller update interval than an hour anyway, we just enforce this as the minimum in all regular updates. This should catch all possible errors leading to a continuous series of requests.

## Motivation and context
#24829 

## How has this been tested?
Manual test stepping through common cases with the debugger on my machine.

## What is the effect on users?
Their broken RPi will not saturate their internet connection any more...

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

lrusak commented on xbmc/addons/RepositoryUpdater.cpp:
```diff
@@ -332,6 +332,18 @@ void CRepositoryUpdater::ScheduleUpdate()
+    // Enforce minimum hold-off time of 1 hour between regular updates - this is especially 
+    // important to handle all sorts of failure cases (e.g., failure to update the add-on database)
+    // that would otherwise lead to an immediate new update attempt and continuous hammering of the servers.
+    delta = std::max(1 * 60 * 60 * 1'000, delta);
```
Can we just use std::chrono types?

yol replied:
no

yol replied:
I'll do another PR for changing the whole file to chrono later, but for omega I want to have it as nonintrusive as possible

lrusak replied:
Boooo 😛

lrusak commented on xbmc/addons/RepositoryUpdater.h:
```diff
@@ -54,12 +54,20 @@ class CRepositoryUpdater : private ITimerCallback, private IJobCallback, public
 
+  enum class UpdateScheduleType
+  {
+    /// Update should be scheduled as the first update after application start or setting change
```
Are these doxygen comments? https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#92-doxygen

ksooo commented:
@yol Your PR is for master, thus v22. For Omega, you need to open backport PR heading for Omega branch.

yol committed:
Enforce minimum interval between add-on repository updates

Repository update can fail in a way that would lead to an
immediate reschedule of the update. A common reason would be
a failure to update the `nextcheck` column of the `repo` table.
This leads to mirror and repository servers getting constantly
hammered.

Since there is no reason to have a smaller update interval than
an hour anyway, we just enforce this as the minimum in all
regular updates. This should catch all possible errors leading
to a continuous series of requests.

Fixes #24829.

neo1973 reviewed (approved):
Lgtm. Tested triggering updates manually and they happen immediately, didn't check for actual waiting for the regular update interval.
